### PR TITLE
`azurerm_route_server` - update resource creation future.

### PR DESCRIPTION
**Pull request by xiaxyi**

### CODE REVIEW

File: internal/services/network/route_server_resource.go [hunk 1]
```diff
@@ -153,7 +158,7 @@ func resourceRouteServerCreateUpdate(d *pluginsdk.ResourceData, meta interface{}
 		ContinuousTargetOccurence: 5,
 		Timeout:                   time.Until(timeout),
 	}
-	_, err := stateConf.WaitForStateContext(ctx)
+	_, err = stateConf.WaitForStateContext(ctx)
```
Comment (tombuildsstuff) [inline internal/services/network/route_server_resource.go:161]: is this still required if we poll on the future?

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
hey @xiaxyi 

Thanks for this PR - taking a look through here it appears that we should be able to remove the extra polling since we're now polling on the future, would you be able to check if that's the case?

Thanks!

File: internal/services/network/route_server_resource.go (hunk 1 above)
Comment (xiaxyi) [reply internal/services/network/route_server_resource.go:161]: Thanks @tombuildsstuff for the comments, yes, this is still needed as we need to get the resource provisioningStatus to ensure the resource is in a ready state.

**Review by katbyte — APPROVED** [review]
LGTM 🦕

**Pull request merged**